src/gui/widgets/StepRecorderWidget.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * StepRecoderWidget.cpp - widget that provide gui markers for step recording
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "StepRecorderWidget.h"
+#include "TextFloat.h"
+#include "embed.h"
+
+StepRecorderWidget::StepRecorderWidget( 
+		QWidget * parent, 
+		const int ppt,    
+		const int margin_top,
+		const int margin_bottom,
+		const int margin_left,
+		const int margin_right) : 
+	QWidget(parent),
+	m_margin_top(margin_top),
+	m_margin_bottom(margin_bottom),
+	m_margin_left(margin_left),
+	m_margin_right(margin_right)
+{
+	const QColor baseColor =  QColor(255, 0, 0);// QColor(204, 163, 0); // Orange
+	m_colorLineEnd   = baseColor.lighter(150);
+	m_colorLineStart = baseColor.darker(120);
+
+	setAttribute(Qt::WA_NoSystemBackground, true);
+	setPixelsPerTact(ppt);
+
+	m_top = m_margin_top;
+	m_left = m_margin_left;
+}
+
+void StepRecorderWidget::setPixelsPerTact(int _ppt )
+{
+	m_ppt = _ppt;
+}
+
+void StepRecorderWidget::setCurrentPosition(MidiTime currentPosition)
+{
+	m_currentPosition = currentPosition;
+}
+
+void StepRecorderWidget::setBottomMargin(const int margin_bottom)
+{
+	m_margin_bottom = margin_bottom;
+}
+
+void StepRecorderWidget::setStartPosition(MidiTime pos)
+{
+	m_curStepStartPos = pos;	
+}
+
+void StepRecorderWidget::setEndPosition(MidiTime pos)
+{
+	m_curStepEndPos = pos;	
+	emit positionChanged( m_curStepEndPos );
+}
+
+void StepRecorderWidget::showHint()
+{
+	TextFloat::displayMessage( tr( "Hint" ),
```

Comment:
According to our coding convention, you shouldn't put whitespaces in parentheses in new files. You can put them in old files for consistency though.